### PR TITLE
Check if a IPv6-only container can be reached

### PR DESCRIPTION
**Pull request by pdostal**
- Related ticket: [poo#134783](https://progress.opensuse.org/issues/134783)
- Verification run:

### CODE REVIEW

File: tests/containers/podman_ipv6.pm [hunk 1]
```diff
@@ -75,6 +76,12 @@ sub run {
     # Test container IPv6 connectivity
     #   there is iptables masquarade so the container appears under the host address
     assert_script_run("podman exec -it test-ipv6 curl -sSf6 -o /dev/null $curl_test https://opensuse.org");
+
+    # Check if a IPv6-only container can be reached
+    $image = 'registry.opensuse.org/opensuse/nginx:latest';
+    script_retry("podman pull $image", timeout => 300, delay => 15, retry => 3);
+    assert_script_run("podman run --name http-ipv6 --network podman-ipv6 --ip6 fd00:c0de:ba5e::3 -d -p 8080:80 $image", timeout => 180);
+    script_retry("curl -v http://127.0.0.1:8080/", delay => 15, retry => 3);
```
Comment (ricardobranco777) [inline tests/containers/podman_ipv6.pm:84]: Maybe it's overkill, but we can add `-4` to be extra sure:

```suggestion
    script_retry("curl -4 -v http://127.0.0.1:8080/", delay => 15, retry => 3);
```

**Review by ricardobranco777 — APPROVED** [review]

File: tests/containers/podman_ipv6.pm (hunk 1 above)
Comment (pdostal) [reply tests/containers/podman_ipv6.pm:84]: Podman by default creates IPv6 NAT. I'll create both `curl -4` and `curl -6` and will let you know.

Comment (pdostal) [reply tests/containers/podman_ipv6.pm:84]: So:
```bash
# curl -4v http://127.0.0.1:8080/; echo StbBp-$?-
*   Trying 127.0.0.1:8080...
* Connected to 127.0.0.1 (127.0.0.1) port 8080
<h1>Welcome to nginx!</h1>
* Connection #0 to host 127.0.0.1 left intact
StbBp-0-
# curl -6v http://127.0.0.1:8080/; echo VMV2j-$?-
* Closing connection
curl: (7) Couldn't connect to server
VMV2j-7-
```
what do you think @ricardobranco777? Is this expected?

Comment (ricardobranco777) [reply tests/containers/podman_ipv6.pm:84]: > So:
> 
> ```shell
> # curl -4v http://127.0.0.1:8080/; echo StbBp-$?-
> *   Trying 127.0.0.1:8080...
> * Connected to 127.0.0.1 (127.0.0.1) port 8080
> <h1>Welcome to nginx!</h1>
> * Connection #0 to host 127.0.0.1 left intact
> StbBp-0-
> # curl -6v http://127.0.0.1:8080/; echo VMV2j-$?-
> * Closing connection
> curl: (7) Couldn't connect to server
> VMV2j-7-
> ```
> 
> what do you think @ricardobranco777? Is this expected?

Yes.  You must use an IPv6 address with `-6`.

Comment (pdostal) [reply tests/containers/podman_ipv6.pm:84]:
```
# curl -6v http://[::1]:8080/; echo Q8DBv-$?-
*   Trying [::1]:8080...
* connect to ::1 port 8080 from ::1 port 49530 failed: Connection timed out
* Failed to connect to ::1 port 8080 after 133479 ms: Couldn't connect to server
* Closing connection
curl: (28) Failed to connect to ::1 port 8080 after 133479 ms: Couldn't connect to server
Q8DBv-28-
```
still ...

Comment (pdostal) [reply tests/containers/podman_ipv6.pm]: Please recheck

File: tests/containers/podman_ipv6.pm [hunk 2]
```diff
@@ -21,7 +22,7 @@ sub _cleanup {
     $podman->cleanup_system_host();
 
     my $registry_ipv4 = script_output('dig +short registry.opensuse.org A | grep -v suse');
-    assert_script_run("iptables -D OUTPUT -d $registry_ipv4 -j DROP");
+    script_run("iptables -D OUTPUT -d $registry_ipv4 -j DROP");
```
Comment (asmorodskyi) [inline tests/containers/podman_ipv6.pm:25]: which means if something goes wrong and cleanup fail we will not notice :thinking:  can you elaborate why you decided to remove failure here ?

Comment (pdostal) [reply tests/containers/podman_ipv6.pm:25]: In general you should test in `run()` not in `post_run_hook()`. But I'll revert this.

**Pull request merged**